include/zephyr/arch/cache.h

```diff
@@ -335,6 +335,20 @@ size_t arch_icache_line_size_get(void);
 
 #endif /* CONFIG_ICACHE || __DOXYGEN__ */
 
+#if CONFIG_CACHE_DOUBLEMAP  || __DOXYGEN__
+bool arch_cache_is_ptr_cached(void *ptr);
+#define cache_is_ptr_cached(ptr) arch_cache_is_ptr_cached(ptr)
+
+bool arch_cache_is_ptr_uncached(void *ptr);
+#define cache_is_ptr_uncached(ptr) arch_cache_is_ptr_uncached(ptr)
+
+void *arch_cache_cached_ptr_get(void *ptr);
+#define cache_cached_ptr(ptr) arch_cache_cached_ptr_get(ptr)
```

Comment:
This seems to be missing some sparse annotations, sparse fails like this:
https://github.com/thesofproject/sof/pull/8824#issuecomment-1920009003
```
include/zephyr/arch/cache.h:345:6: error: symbol 'arch_cache_cached_ptr_get' redeclared with different type (different address spaces):
include/zephyr/arch/cache.h:348:6: error: symbol 'arch_cache_uncached_ptr_get' redeclared with different type (incompatible argument 1 (different address spaces)):
include/zephyr/arch/cache.h:345:6: error: symbol 'arch_cache_cached_ptr_get' redeclared with different type (different address spaces):
include/zephyr/arch/cache.h:348:6: error: symbol 'arch_cache_uncached_ptr_get' redeclared with different type (incompatible argument 1 (different address spaces)):
include/zephyr/arch/cache.h:345:6: error: symbol 'arch_cache_cached_ptr_get' redeclared with different type (different address spaces):
include/zephyr/arch/cache.h:348:6: error: symbol 'arch_cache_uncached_ptr_get' redeclared with different type (incompatible argument 1 (different address spaces)):

```

Reply:
addressed.

Reply:
Much fewer sparse warnings now but still a good few: https://github.com/thesofproject/sof/pull/8824#pullrequestreview-1855124913

Running sparse locally is very easy thanks to @tejlmand's work:

1. Download and compile sparse with a simple `make`
2. add `sparse` to PATH
3. west build .... -DZEPHYR_SCA_VARIANT=sparse

Reply:
> not getting any of those when I build SOF for example using sparse as SCA

Detailed reproduction steps:

```shell
cd zephyr
git fetch origin 3284d869565ef6a254868853e1102cdff438816e
git checkout     3284d869565ef6a254868853e1102cdff438816e
west update
west build -p -b intel_adsp_ace15_mtpm ../modules/audio/sof/app/ -DZEPHYR_SCA_VARIANT=sparse |& tee sparse.log

../modules/audio/sof/scripts/parse_sparse_output.sh < sparse.log 

...
zephyr/include/zephyr/cache.h:511:16: warning: incorrect type in return expression (different address spaces)
zephyr/include/zephyr/cache.h:511:16: warning: incorrect type in return expression (different address spaces)
zephyr/include/zephyr/cache.h:511:16: warning: incorrect type in return expression (different address spaces)
zephyr/include/zephyr/cache.h:511:16: warning: incorrect type in return expression (different address spaces)
zephyr/include/zephyr/cache.h:511:16: warning: incorrect type in return expression (different address spaces)
...
zephyr/include/zephyr/cache.h:535:16: warning: incorrect type in argument 1 (different address spaces)
zephyr/include/zephyr/cache.h:535:16: warning: incorrect type in argument 1 (different address spaces)
zephyr/include/zephyr/cache.h:535:16: warning: incorrect type in argument 1 (different address spaces)

...
```


There are still some warnings left even after `git checkout HEAD~3 && west update` (I will look at them now) but the address warnings above in `cache.h` are caused by this PR.

Reply:
> There are still some warnings left even after git checkout HEAD~3 && west update (I will look at them now)

That's just because I forgot that we must always use `-DCONFIG_LOG_USE_VLA=n` when using sparse. I forgot because the final outcome and messages when forgetting it is `too many warnings!` which isn't very useful.

To summarize; when using `-DCONFIG_LOG_USE_VLA=n` + `modules/audio/sof/scripts/parse_sparse_output.sh`:
- before this PR: zero address space warning
- with this PR: about 1 screen long of "different address space" warnings.